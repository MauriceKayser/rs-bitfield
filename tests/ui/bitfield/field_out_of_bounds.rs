//! Assertion failure: _FIELD_0_EXCEEDS_THE_BIT_FIELD_SIZE

extern crate alloc;

#[bitfield::bitfield(size)]
struct BitField(#[field(250, 2)] Field); // Can only store bits between 0 - (sizeof(usize) * 8).

#[derive(Debug, bitfield::Field)]
#[repr(u8)]
enum Field {
    F1 = 1,
    F2,
    F3
}

fn main() {}