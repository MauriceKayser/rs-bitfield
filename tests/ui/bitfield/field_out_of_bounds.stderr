error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/field_out_of_bounds.rs:5:1
  |
5 | #[bitfield::bitfield(size)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Field 0 exceeds the bitfield size', $DIR/tests/ui/bitfield/field_out_of_bounds.rs:5:1
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
